bsip-00XX Dynamic market fees.md

```diff
@@ -0,0 +1,183 @@
+Up for discussion is - what to take as a day 1. 1st trade of this asset? 1st trade minus 30 days?  Account open date? 
+This way we have a normal fair average volume if we figure out the fair beginning interval date.
+However, if a trader started trading this asset a few years ago he might be in a significant disadvantage.
+
+- **Average trading volume over the last 30 days - sliding window**.
+This way it is a bit clearer how to calculate. 
+However there might be extreme but real cases where this kind of calculation might not be fair: 
+a. when a person who usually trades high volumes throughout the year goes to vacation for 30 days. Suddenly all his discount is gone when he comes back
+b.   When another person comes in and makes 1 or 2 high volume trades while trading really low volumes before that. Suddenly this person might have a big discount. 
+30-day volume 
+This mehod gives enough information to figure out last 30-days traders habits with much lower memory requirements but slightly less precision than the 30-day average sliding window.
+Average trading volume over the last 365 days (a year) is a variation of this method.
+
+- **30-day volume method**
+This is quite similar to the 30-day average method and shares it's advantages and drawbacks.
+
+# Technical Specifications
+
+We suggest using average total volume for a particular asset. However, it's open for the discussion.
+
+## Statistics Storage
+
+There are two options for storing account statistics for trading:
+
+1. expand class account_statistics_object by adding a map of trade_statistics_object for each asset
+2. create a new type trade_statistics_object and a new index
+
+
+## Statistics Update
+Statistics updates during fill_order. 
+We need to update statistics for the user and for the specific asset at the same time.
+
+ 
+
+## Calculations
+
+### Average since Day 1
+
+**Trader Activity** is estimated as average trading volume of a user for a given asset for 1 day.
+
+`Trader_activity = total_volume / (Now - first_start_date) //(in days).`
+
+This way we accumulate total bought volume for each user for each asset and average out daily volumes for his entire term of trading of this asset (per day).
+
+Volume discounts are estimated based on the average daily `Trader_Activity  * 30 days.`
+
+_Trade statistics object_
+
+
+```
+class trade_statistics_object
+{
+
+account_id_type owner;
+asset_id_type asset;
+timestamp first_trade_date; //first trade date for this asset
+share_type total_volume; //total bought asset volume
+};
+```
+
+### 30-day sliding window average 
+
+In **trade_statistics_object** we need to store the queue of volumes aggregated by 1 day. In other words, for each asset a user should have a queue of 30  daily volumes.
+
+During the first maintenance interval of the day, yesterday's volume will be added to the queue and the  oldest volume will be deleted (for each asset even for assets where there was no trade done).
```

Comment:
a.k.a. 30 day FIFO (first-in first-out) buffer

Reply:
queue = FIFO